7_drivers_munge/src/GCM_driver_utils.R

```diff
@@ -78,18 +78,10 @@ construct_grid <- function(cellsize, nx, ny, xmin, ymin, crs) {
   return(grid_sf)
 }
 
-# TODO: scale up. For now, this just subsets the lakes
-# to 5 in WI. Will want to scale up to all of MN and then
-# try for every lake in centroid_lakes_sf Once we scale
-# up completely, we shouldn't need this function.
+# TODO: Fancier version. This is a very rudimentary way to get all MN lakes.
 subset_lake_centroids <- function(centroids_sf) {
-
-  set.seed(19) # Same subset of 5 every time
-  wi_sf <- st_as_sf(maps::map('state', 'wisconsin', plot=FALSE, fill=TRUE))
-
   centroids_sf %>%
-    st_intersection(wi_sf) %>%
-    sample_n(5)
+    st_intersection(st_as_sf(maps::map('state', 'minnesota', plot=FALSE, fill=TRUE)))
```

Comment:
I captured this "to do" as part of #271 but will need to prioritize it (at least for MN) soon. Do you see this as a polygon-to-polygon matching step to figure out which lakes are in which state using `canonical_lakes_sf` and then some spatial data for the states (like from Census Bureau)?

Reply:
Yes, I think census counties are the appropriate unit (which could be dissolved into states). In mntoha, this was: https://github.com/USGS-R/mntoha-data-release/blob/master/1_spatial.yml#L58

There is a question about whether you use the centroids (_within state_) or polygon (_overlaps part of state_)

In MNTOHA, the polygon overlap resulted in some lakes that belong to multiple states, since they overlap the borders (same with county ownership of a lake).

Reply:
Hmmm ok. I am trying to think of the implications for why we want to use this. We want to 1) be able to _subset data_ to see what we have for lakes _in_ certain states, and 2) want to be able to _summarize available data_ by state. If we decided to do polygon matching and allowed some lakes to belong to multiple states, we wouldn't need to remember that in downstream steps, e.g. if we wanted to model MN & WI we couldn't just filter by states to get the sites since there could be duplicates. We would need to also do a `unique()` before modeling. So would just need to note that next to the target.

Reply:
Not sure that we'd want to go this route, but we could have a table with a row for each lake and lake_id, then columns for each state, with a binary 0 or 1 for whether that lake is present in that state. Then for modeling multiple lakes, you could use AND or OR statements for filtering to get the subset of lake ids that you want. That way we wouldn't have duplicate rows.

Reply:
See #278